src/app/accounts/login.component.ts

```diff
@@ -55,4 +63,31 @@ export class LoginComponent extends BaseLoginComponent {
             modal.close();
         });
     }
+
+    async launchSsoBrowser() {
+        // Generate necessary sso params
+        const passwordOptions: any = {
+            type: 'password',
+            length: 64,
+            uppercase: true,
+            lowercase: true,
+            numbers: true,
+            special: false,
+        };
+        const state = await this.passwordGenerationService.generatePassword(passwordOptions);
+        let ssoCodeVerifier = await this.passwordGenerationService.generatePassword(passwordOptions);
+        const codeVerifierHash = await this.cryptoFunctionService.hash(ssoCodeVerifier, 'sha256');
+        const codeChallenge = Utils.fromBufferToUrlB64(codeVerifierHash);
+
+        // Build URI
+        const webUrl = this.environmentService.webVaultUrl == null ? 'https://vault.bitwarden.com' :
+            this.environmentService.webVaultUrl;
+        const clientId = 'desktop';
+        const ssoRedirectUri = 'bitwarden://sso-callback';
+
+        // Launch browser
+        this.platformUtilsService.launchUri(webUrl + '/#/sso?clientId=' + clientId +
+            '&redirectUri=' + encodeURIComponent(ssoRedirectUri) +
+            '&state=' + state + '&codeChallenge=' + codeChallenge);
+    }
```

Comment:
Does this work? I don't see where you are persisting the code verifier or the state to check on callback.

Reply:
Did I miss something? I don't recall seeing this in the dc or cli implementation?

Reply:
@kspearrin Per our conversation - I've updated this over in the [jslib PR](https://github.com/bitwarden/jslib/pull/144)